src/helpers/overrides.js

```diff
@@ -206,13 +211,53 @@ export function useOverrides(
   return React.useMemo(
     () =>
       // eslint-disable-next-line flowtype/no-weak-types
-      Object.keys(defaults).reduce<{[string]: [React.ComponentType<any>, {}]}>(
-        (obj, key) => {
-          obj[key] = getOverrides(overrides[key], defaults[key]);
-          return obj;
-        },
-        {},
-      ),
+      Object.keys(defaults).reduce<{
+        // eslint-disable-next-line flowtype/no-weak-types
+        [string]: [React.ComponentType<any>, {}],
+      }>((obj, key) => {
+        obj[key] = getOverrides(overrides[key], defaults[key]);
+        return obj;
+      }, {}),
     [overrides],
   );
 }
+
+export function withOverrides<Config, Instance>(
+  WrappedComponent: React.AbstractComponent<Config, Instance>,
+  componentName: string,
+  // eslint-disable-next-line flowtype/no-weak-types
+): React.AbstractComponent<Config, Instance> {
+  // eslint-disable-next-line flowtype/no-weak-types
+  class WithOverrides extends React.Component<{
+    ...Config,
+    // eslint-disable-next-line flowtype/no-weak-types
+    forwardedRef: ?any,
+    overrides?: OverridesT,
+  }> {
+    static contextType = ThemeContext;
+    constructor(props) {
+      super(props);
+    }
+
+    render() {
+      // Grab forwardedRef from props and pass it down
+      const {forwardedRef} = this.props;
+      // mergeOverrides from themeContext and props
+      const overrides = mergeOverrides(
+        this.context.overrides?.[componentName],
```

Comment:
this causes a flow error https://buildkite.com/uberopensource/baseweb/builds/18428#9050979e-44d2-49d6-b921-3d25a5063830/128-132